### ISSUE TITLE
stack header must stay on top

### ISSUE THREAD

**Issue by cblp** [tweag/ormolu#393]
Stack expects its header in the beginning of the file or just after shebang

Examples that must be idempotent (modulo whitespace):
```hs
-- stack runhaskell
{-# LANGUAGE OverloadedStrings #-}
main = pure ()
```

```hs
#!/usr/bin/env stack
-- stack runhaskell
{-# LANGUAGE OverloadedStrings #-}
main = pure ()
```

**Comment by mrkkrp** [comment]
What does this header actually do and why this cannot be done by just using shebang?

**Comment by cblp** [comment]
This runs interpreter installed by stack https://docs.haskellstack.org/en/stable/GUIDE/#script-interpreter

Just shebang has the same problem. Should I open another ticket?

**Comment by mrkkrp** [comment]
I fixed the shebang thing recently, see #377.

**Comment by cblp** [comment]
Well, but both of these examples are not idempotent.

**Comment by mrkkrp** [comment]
I believe I fixed #377 properly now. This issue should be easy to fix now.
